inspirehep/modules/records/receivers.py

```diff
@@ -178,12 +178,35 @@ def enhance_after_index(sender, json, *args, **kwargs):
     populate_affiliation_suggest(sender, json, *args, **kwargs)
     populate_author_count(sender, json, *args, **kwargs)
     populate_authors_full_name_unicode_normalized(sender, json, *args, **kwargs)
+    populate_author_suggest(sender, json, *args, **kwargs)
     populate_earliest_date(sender, json, *args, **kwargs)
     populate_inspire_document_type(sender, json, *args, **kwargs)
     populate_name_variations(sender, json, *args, **kwargs)
     populate_title_suggest(sender, json, *args, **kwargs)
 
 
+def populate_author_suggest(sender, json, *args, **kwargs):
+    """Populate the ``author_suggest`` field of Authors records."""
+    if 'authors.json' not in json.get('$schema'):
+        return
+
+    author_paths = [
+        'name.preferred_name',
+        'name.value',
+        'native_name',
+        'other_names',
+        'previous_names',
```

Comment:
seems OK, note that this is going to change with https://github.com/inspirehep/inspire-schemas/pull/182, where all those names will be under `name`.